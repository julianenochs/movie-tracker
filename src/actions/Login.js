const addUser = ({ email, password }) => ({
    type: 'ADD_USER',
    email,
    password
});