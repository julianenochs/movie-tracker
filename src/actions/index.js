const addUser = ({ email, password }) => ({
    type: 'ADD_USER',
    email,
    password
});

export const addMovies = movies => ({
  type: 'ADD_MOVIES',
  movies,
});