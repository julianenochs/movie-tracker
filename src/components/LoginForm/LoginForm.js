import React, { Component } from 'react';
import { connect } from 'react-redux';

class LoginForm extends Component {
    constructor() {
        super();
        this.state = { 
            email: '',
            password: '' 
        }
    }

    handleSubmit = () => {
        this.setState({ email: '', password: ''})
    }

    handleChange = (e) => {
        this.setState({ [e.target.name] : e.target.value })
    }

    render() {
        return(
            <section>
                <form onSubmit={this.handleSubmit}>
                    <input
                        name='email'
                        value={this.state.email}
                        placeholder='Your email here'
                        onChange={this.handleChange} />
                    <input 
                        name='password'
                        value={this.state.password}
                        placeholder='Password'
                        onChange={this.handleChange} />
                        <button>Login</button>
                </form>
            </section>
        )
    }
}

export default LoginForm;