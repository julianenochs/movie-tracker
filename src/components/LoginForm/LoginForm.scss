$button-color: #004263;
$button-shadow: #003149;

.form {
  display: flex;
  flex-direction: column;
  width: 65%;
}

form {
  align-items: center;
  display: flex;
  justify-content: space-around;
}

.login__input {
  padding-left: 10px;
  height: 40px;
  font-size: 1em;
  margin: 20px;
  width: 300px;
}

.login__button {
  background-color: $button-color;
  border: none;
  border-radius: 5px;
  box-shadow: 4px 4px $button-shadow;
  color: white;
  font-family: 'Barlow Condensed', sans-serif;
  font-size: 1.5em;
  height: 40px;
  letter-spacing: 5px;
  width: 120px;
}

.error__message {
  color: white;
}