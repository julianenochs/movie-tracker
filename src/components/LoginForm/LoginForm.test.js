import React from 'react';
import { shallow } from 'enzyme';
import { LoginForm, mapStateToProps, mapDispatchToProps } from './LoginForm';
import { login } from '../../actions';

describe('LoginForm', () => {
  let wrapper;
  const mockMovies = [
    {
      popularity: 471.315,
      vote_count: 4211,
      video: false,
      poster_path: '/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg',
      id: 475557,
      adult: false,
      backdrop_path: '/n6bUvigpRFqSwmPp1m2YADdbRBc.jpg',
      original_language: 'en',
      original_title: 'Joker',
      title: 'Joker',
      vote_average: 8.6,
      overview:
        'During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.',
      release_date: '2019-10-04',
      isFavorite: false
    }
  ];
  const mockUpdateUserInfo = jest.fn();
  const mockUpdateIsLoggedIn = jest.fn();
  const mockUpdateError = jest.fn();
  const mockUpdateUser = jest.fn();
  const mockResetError = jest.fn();
  const mockUpdateFavorites = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <LoginForm
        user={'Peerat Test'}
        tempUser={'Peerat Test'}
        isLoggedIn={true}
        error={''}
        movies={mockMovies}
        updateUserInfo={mockUpdateUserInfo}
        updateIsLoggedIn={mockUpdateIsLoggedIn}
        updateError={mockUpdateError}
        updateUser={mockUpdateUser}
        resetError={mockResetError}
        updateFavorites={mockUpdateFavorites}
      />
    );
  });

  it('should match the LoginForm Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleLogin when the button is clicked', () => {});

  describe('mapStateToProps', () => {
    it('should return an object with a user object', () => {
      const mockState = {
        user: { email: 'jujube@email.com', password: 'password' },
        tempUser: {
          name: 'Juju',
          email: 'jujube@email.com',
          password: 'password'
        }
      };
      const expected = {
        user: { email: 'jujube@email.com', password: 'password' },
        tempUser: {
          name: 'Juju',
          email: 'jujube@email.com',
          password: 'password'
        }
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});
