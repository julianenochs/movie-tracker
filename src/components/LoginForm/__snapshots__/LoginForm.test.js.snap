// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the LoginForm Snapshot 1`] = `ShallowWrapper {}`;
